core/src/main/scala/akka/kafka/scaladsl/Producer.scala

```diff
@@ -72,6 +74,35 @@ object Producer {
       .mapAsync(settings.parallelism)(_.message.passThrough.commitScaladsl())
       .toMat(Sink.ignore)(Keep.right)
 
+  /**
+   * Sink that is aware of the [[ConsumerMessage.TransactionalMessage#PartitionOffset]] from a [[Consumer]].  It will
+   * initialize, begin, produce, and commit the consumer offset as part of a transaction.
+   */
+  def transactionalSink[K, V](
+    settings: ProducerSettings[K, V],
+    transactionalId: String
+  ): Sink[Message[K, V, ConsumerMessage.PartitionOffset], Future[Done]] = {
+    transactionalFlow(settings, transactionalId)
+      .toMat(Sink.ignore)(Keep.right)
+  }
+
+  /**
+   * Publish records to Kafka topics and then continue the flow.  The flow should only used with a [[Consumer]] that
+   * emits a [[ConsumerMessage.TransactionalMessage]].  The flow requires a unique `transactional.id` across all app
+   * instances.  The flow will override producer properties to enable Kafka exactly once transactional support.
+   */
+  def transactionalFlow[K, V](settings: ProducerSettings[K, V], transactionalId: String): Flow[Message[K, V, ConsumerMessage.PartitionOffset], Result[K, V, ConsumerMessage.PartitionOffset], NotUsed] = {
+    require(transactionalId != null && transactionalId.length > 0, "You must define a Transactional id.")
+
+    val txProducerSettings = settings.withProperties(
+      ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG -> true.toString,
+      ProducerConfig.TRANSACTIONAL_ID_CONFIG -> transactionalId,
+      ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION -> "1"
```

Comment:
`1.toString` ?  (Or even better, consider defining default values  as constants)

Reply:
This will never change (at least not with current releases of Kafka) and is only used with `transactionalSink` and `transactionalFlow`, so I think it makes sense to keep it close to the implementation.

Reply:
1.toString +1